tests/unit/docs/__init__.py

```diff
@@ -0,0 +1,178 @@
+                'serviceFullName': 'AWS MyService',
+                'protocol': 'query'
+            },
+            'operations': {
+                'SampleOperation': {
+                    'name': 'SampleOperation',
+                    'input': {'shape': 'SampleOperationInputOutput'},
+                    'output': {'shape': 'SampleOperationInputOutput'}
+                }
+            },
+            'shapes': {
+                'SampleOperationInputOutput': {
+                    'type': 'structure',
+                    'members': OrderedDict()
+                },
+                'String': {
+                    'type': 'string'
+                }
+            }
+        }
+
+        self.waiter_json_model = {
+            "version": 2,
+            "waiters": {
+                "SampleOperationComplete": {
+                    "delay": 15,
+                    "operation": "SampleOperation",
+                    "maxAttempts": 40,
+                    "acceptors": [
+                        {"expected": "complete",
+                         "matcher": "pathAll",
+                         "state": "success",
+                         "argument": "Biz"},
+                        {"expected": "failed",
+                         "matcher": "pathAny",
+                         "state": "failure",
+                         "argument": "Biz"}
+                    ]
+                }
+            }
+        }
+
+        self.paginator_json_model = {
+            "pagination": {
+                "SampleOperation": {
+                    "input_token": "NextResult",
+                    "output_token": "NextResult",
+                    "limit_key": "MaxResults",
+                    "result_key": "Biz"
+                }
+            }
+        }
+
+    def build_models(self):
+        self.service_model = ServiceModel(self.json_model)
+        self.operation_model = OperationModel(
+            self.json_model['operations']['SampleOperation'],
+            self.service_model
+        )
+
+    def add_shape(self, shape):
+        shape_name = list(shape.keys())[0]
+        self.json_model['shapes'][shape_name] = shape[shape_name]
+
+    def add_shape_to_params(self, param_name, shape_name, documentation=None,
+                            is_required=False):
+        params_shape = self.json_model['shapes']['SampleOperationInputOutput']
+        member = {'shape': shape_name}
+        if documentation is not None:
+            member['documentation'] = documentation
+        params_shape['members'][param_name] = member
+
+        if is_required:
+            required_list = params_shape.get('required', [])
+            required_list.append(param_name)
+            params_shape['required'] = required_list
+
+    def assert_contains_line(self, line):
+        contents = self.doc_structure.flush_structure().decode('utf-8')
+        self.assertIn(line, contents)
+
+    def assert_contains_lines(self, lines):
```

Comment:
Can we update this to at assert the lines are in order?  Right now this will also pass with any order.